Automattic/jetpack
# Proposal: switch email sharing to use mailto links

This PR constitutes initial steps to have the Share by Email functionality use a client-side `mailto:` link instead of a form submission. This should prevent this feature from being a source of spam, while still allowing users to share links with other people. A key change is that the actual email transmission will no longer occur from the WordPress hosting environment, but will come directly from the user.

#### Changes proposed in this Pull Request:

* This PR updates Email sharing to use a client-side `mailto:` link that shares a link via the user's mail client instead of a form submission that then triggers an outbound email from the WordPress host
* Due to this change, we now default to enabling Share by Email, and don't filter it being enabled as it is no longer a risk to the server infrastructure, nor does it depend on Akismet being enabled
* It is worth noting that the new implementation still sends data to the server for the purposes of stat tracking, but no longer sends any data through Akismet, and no longer collects any email addresses

#### Jetpack product discussion
There was some background discussion on this topic in p58i-cjq-p2 and p1HpG7-fuT-p2, but this specific approach wasn't discussed in depth.
I started exploring this for WordPress.com in D79210-code, which was the starting point for this PR.
Dedicated discussion is in p1HpG7-fLv-p2.

#### Does this pull request change what data or activity we track or use?
No, it does not. If anything, it will eventually _reduce_ the amount of data we use, as we will no longer be passing any user-specific data through to the server or Akismet for sharing by email.

#### Testing instructions:

##### Initial Setup

* Create a new site for testing, and ensure that you have the Jetpack, Jetpack Beta and Gutenberg plugins installed.
* Open up wp-admin for your site, and navigate to the **Jetpack** menu item.
* Work through the Jetpack connection flow to set up Jetpack on the free plan.
* Navigate to **Jetpack** -> **Jetpack Beta**, and switch the main Jetpack plugin to use the branch for this PR: `try/switch-email-sharing-to-use-mailto-links`
* Navigate to **Jetpack** -> **Settings**, and then click on the "Sharing" tab
* In the "Sharing buttons" section, enable the "Add sharing buttons to your posts and pages" toggle
* When that is enabled and you've seen a success notice, click on the "Configure your sharing buttons" link -- this should open a new tab on the WordPress.com **Marketing** -> **Tools** page with the "Sharing Buttons" tab active
* Ensure that the "Posts" option is checked under the "Show like and sharing buttons on" section
* Click on the "Edit sharing buttons" button, which should show a number of sharing options, including the "Email" button
* Click on the "Email" button to enable that sharing option
* Click on "Save changes" to save the updated settings
* Navigate to **Jetpack** -> **My Jetpack**
* Find the "Anti-Spam" card, and click on the "Activate" button
* Click on the "Manage" button that replaces the "Activate" button, and perform any steps needed to enable Akismet
  - This step is a bit fuzzy as I worked through an issue with my Akismet subscription!
* Ensure that you have at least one post for your site -- create a new one if needed
* Ensure you have an email client configured for mailto links. It can be Gmail or another email client.

Ideally, we want to test the following pages and sharing approaches using multiple browsers, including any mobile web browsers you can access.

##### Testing the new mailto approach

This code defaults to the new mailto approach, so we'll test that first.

* Navigate to a post on your site - you should be viewing only the full post.
* Verify that you can see the "Email" sharing button on the page.
* Mouse over the Email button and verify that the title mentions emailing a link to a friend.
* Open your developer console and switch to the Network tab
* Click on the Email button
* Verify that you see an email created in your preferred mail client. The email should include the post title in the subject, and a link to the post in the email body.
* Also verify that an HTTP request was sent to the back end to track the submission. (You can add debugging/logging to `Share_Email::process_request()` if you want to verify the submission. The primary reason for this submission is to continue tracking these shares.)
* Click the Email button at least two more times - verify that after the third click, you see an informational message appear below the sharing buttons -- the message should make it clear that you might need to configure your email client, or compose an email manually (see below for screenshot)
<img width="706" alt="screenshot_multi_share_error" src="https://user-images.githubusercontent.com/3376401/169541250-02c25207-2b76-4176-a4ee-5ee18cb6a727.png">
